### PR TITLE
Fix indexing bugs with tracks

### LINKED ISSUE #62: Cells lose highlight on last frame in track
## Description

On the last frame in a track, cell tracks can still be selected but won't be highlighted on the canvas.

## Expected Behavior

Cells should be highlighted for every frame that they're visible.

## Reproduction

Skip to frame 569 and try clicking any cell.
http://dev-aics-dtp-001.corp.alleninstitute.org/nucmorph-colorizer/dist/index.html?dataset=Mama%20Bear&feature=NUC_shape_volume_lcc&track=8918&t=466
Go to the above link and advance the frame by one.

### PR DESCRIPTION
**Pull request by toloudis**
Problem
=======
Derek has been asking to see his data from EMT project in this viewer.
In developing the script to convert his data, we discovered some indexing bugs.  This also fixes #62 ...

Solution
========
Bug fixes to the front end with big help from @ShrimpCryptid.  
Bonus: add a work-in-progress script to convert Derek's EMTImmunostaining data for presentation in the colorizer. 

Future work: There will probably be a followup pr with some cleanup to the data prep script.  Please consider the typescript fixes to be the main point of this PR.

## Type of change

* Bug fix (non-breaking change which fixes an issue)
* New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: scripts/timelapse-colorizer-data/generate_data_2.py
Comment (ShrimpCryptid) [inline scripts/timelapse-colorizer-data/generate_data_2.py:1]: If we need to write another data conversion script again, I'll probably want to move all the string labels for common tasks (like getting the X/Y of centroids, tracks, etc.) into constants at the start of the file, so they can be changed a little faster. This isn't something to do for this file, just a thought for later.

**Review by ShrimpCryptid — APPROVED** [review]
Looks good! Thank you for making these fixes to the track and canvas. I added a few comments but most don't need addressing.

File: scripts/timelapse-colorizer-data/generate_data_2.py
Comment (toloudis) [reply scripts/timelapse-colorizer-data/generate_data_2.py:1]: agree!

File: src/colorizer/Track.ts [hunk 1]
```diff
@@ -33,17 +33,18 @@ export default class Track {
...
       return -1;
     }
     return this.ids[index];
   }
 
   length(): number {
-    return this.times.length;
+    // tracks may have gaps (missing or elided data) in their list of times.
+    // So since the times are pre-sorted above, the length of the track is the difference between the last time and the first time.
+    // note that a track with only one time in it will report length 0
```
[the remark this answers is not in the document]
Comment (toloudis) [reply src/colorizer/Track.ts:47]: I'm not sure I tested everything but basically things seem functional.

**Pull request merged**